{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: seaborn in c:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (0.11.1)\n",
      "Requirement already satisfied: matplotlib>=2.2 in c:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from seaborn) (3.4.1)\n",
      "Requirement already satisfied: pandas>=0.23 in c:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from seaborn) (1.2.4)\n",
      "Requirement already satisfied: numpy>=1.15 in c:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from seaborn) (1.19.5)\n",
      "Requirement already satisfied: scipy>=1.0 in c:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from seaborn) (1.6.3)"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "WARNING: You are using pip version 21.3; however, version 21.3.1 is available.\n",
      "You should consider upgrading via the 'c:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\python.exe -m pip install --upgrade pip' command.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Requirement already satisfied: pyparsing>=2.2.1 in c:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from matplotlib>=2.2->seaborn) (2.4.7)\n",
      "Requirement already satisfied: python-dateutil>=2.7 in c:\\users\\shm22004\\appdata\\roaming\\python\\python39\\site-packages (from matplotlib>=2.2->seaborn) (2.8.1)\n",
      "Requirement already satisfied: cycler>=0.10 in c:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from matplotlib>=2.2->seaborn) (0.10.0)\n",
      "Requirement already satisfied: pillow>=6.2.0 in c:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from matplotlib>=2.2->seaborn) (8.2.0)\n",
      "Requirement already satisfied: kiwisolver>=1.0.1 in c:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from matplotlib>=2.2->seaborn) (1.3.1)\n",
      "Requirement already satisfied: pytz>=2017.3 in c:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from pandas>=0.23->seaborn) (2021.1)\n",
      "Requirement already satisfied: six in c:\\users\\shm22004\\appdata\\roaming\\python\\python39\\site-packages (from cycler>=0.10->matplotlib>=2.2->seaborn) (1.16.0)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ERROR: Could not find a version that satisfies the requirement upgrade (from versions: none)\n",
      "ERROR: No matching distribution found for upgrade\n",
      "WARNING: You are using pip version 21.3; however, version 21.3.1 is available.\n",
      "You should consider upgrading via the 'c:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\python.exe -m pip install --upgrade pip' command.\n"
     ]
    }
   ],
   "source": [
    "%pip install seaborn\n",
    "%pip install upgrade pip "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ERROR: Could not find a version that satisfies the requirement requirement (from versions: none)\n",
      "ERROR: No matching distribution found for requirement\n",
      "WARNING: You are using pip version 21.3; however, version 21.3.1 is available.\n",
      "You should consider upgrading via the 'c:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\python.exe -m pip install --upgrade pip' command.\n"
     ]
    }
   ],
   "source": [
    "%pip install requirement"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ERROR: Could not find a version that satisfies the requirement upgrade (from versions: none)\n",
      "ERROR: No matching distribution found for upgrade\n",
      "WARNING: You are using pip version 21.3; however, version 21.3.1 is available.\n",
      "You should consider upgrading via the 'c:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\python.exe -m pip install --upgrade pip' command.\n"
     ]
    }
   ],
   "source": [
    "%pip install upgrade pip"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: jupyterlab_widgets in c:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (1.0.0)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "WARNING: You are using pip version 22.0.3; however, version 22.0.4 is available.\n",
      "You should consider upgrading via the 'C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\python.exe -m pip install --upgrade pip' command.\n"
     ]
    }
   ],
   "source": [
    "#!pip install tensorflow_model_analysis==0.34.0\n",
    "#!jupyter labextension install tensorflow_model_analysis@0.34.0\n",
    "%pip install Jupyter_widgets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting tfx\n",
      "  Using cached tfx-0.23.1-py3-none-any.whl (1.7 MB)\n",
      "Collecting docker<5,>=4.1\n",
      "  Using cached docker-4.4.4-py2.py3-none-any.whl (147 kB)\n",
      "Collecting tensorflow-model-analysis<0.24,>=0.23\n",
      "  Using cached tensorflow_model_analysis-0.23.0-py3-none-any.whl (1.6 MB)\n",
      "Collecting click<8,>=7\n",
      "  Using cached click-7.1.2-py2.py3-none-any.whl (82 kB)\n",
      "Requirement already satisfied: tensorflow!=2.0.*,!=2.1.*,!=2.2.*,<3,>=1.15.2 in c:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from tfx) (2.5.0rc2)\n",
      "Collecting pyarrow<0.18,>=0.17\n",
      "  Using cached pyarrow-0.17.1.tar.gz (2.6 MB)\n",
      "  Installing build dependencies: started\n",
      "  Installing build dependencies: finished with status 'error'\n",
      "  Using cached pyarrow-0.17.0.tar.gz (3.7 MB)\n",
      "  Installing build dependencies: started\n",
      "  Installing build dependencies: finished with status 'error'\n",
      "Collecting tfx\n",
      "  Using cached tfx-0.23.0-py3-none-any.whl (1.7 MB)\n",
      "  Using cached tfx-0.22.2-py3-none-any.whl (1.5 MB)\n",
      "Collecting apache-beam[gcp]<3,>=2.21\n",
      "  Using cached apache-beam-2.33.0.zip (2.6 MB)\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'done'\n",
      "Collecting tfx\n",
      "  Using cached tfx-0.22.1-py3-none-any.whl (1.5 MB)\n",
      "  Using cached tfx-0.22.0-py3-none-any.whl (1.5 MB)\n",
      "  Using cached tfx-0.21.5-py3-none-any.whl (1.1 MB)\n",
      "  Using cached tfx-0.21.4-py3-none-any.whl (1.1 MB)\n",
      "  Using cached tfx-0.21.3-py3-none-any.whl (1.1 MB)\n",
      "  Using cached tfx-0.21.2-py3-none-any.whl (1.1 MB)\n",
      "Requirement already satisfied: jinja2<3,>=2.7.3 in c:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from tfx) (2.11.3)\n",
      "  Using cached tfx-0.21.1-py3-none-any.whl (1.1 MB)\n",
      "  Using cached tfx-0.21.0-py3-none-any.whl (1.0 MB)\n",
      "  Using cached tfx-0.15.0-py3-none-any.whl (504 kB)\n",
      "Requirement already satisfied: protobuf<4,>=3.7 in c:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from tfx) (3.15.8)\n",
      "  Using cached tfx-0.14.0-py3-none-any.whl (384 kB)\n",
      "Collecting tensorflow-model-analysis<0.15,>=0.14\n",
      "  Downloading tensorflow_model_analysis-0.14.0-py3-none-any.whl (777 kB)\n",
      "Collecting tfx\n",
      "  Using cached tfx-0.13.0-py2.py3-none-any.whl (212 kB)\n",
      "Collecting tensorflow-transform<0.14,>=0.13\n",
      "  Using cached tensorflow-transform-0.13.0.tar.gz (173 kB)\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'done'\n",
      "\n",
      "The conflict is caused by:\n",
      "    tfx 0.23.1 depends on pyarrow<0.18 and >=0.17\n",
      "    tfx 0.23.0 depends on tensorflow-data-validation<0.24 and >=0.23\n",
      "    tfx 0.22.2 depends on ml-metadata<0.23 and >=0.22.1\n",
      "    tfx 0.22.1 depends on ml-metadata<0.23 and >=0.22.1\n",
      "    tfx 0.22.0 depends on tfx-bsl<0.23 and >=0.22\n",
      "    tfx 0.21.5 depends on tensorflow-data-validation<0.22 and >=0.21.4\n",
      "    tfx 0.21.4 depends on tensorflow-data-validation<0.22 and >=0.21.4\n",
      "    tfx 0.21.3 depends on tensorflow-data-validation<0.22 and >=0.21.4\n",
      "    tfx 0.21.2 depends on tfx-bsl<0.22 and >=0.21.3\n",
      "    tfx 0.21.1 depends on tensorflow-data-validation<0.22 and >=0.21.4\n",
      "    tfx 0.21.0 depends on tfx-bsl<0.22 and >=0.21\n",
      "    tfx 0.15.0 depends on tfx-bsl<0.16 and >=0.15.3\n",
      "    tfx 0.14.0 depends on ml-metadata<0.15 and >=0.14\n",
      "    tfx 0.13.0 depends on ml-metadata<0.14 and >=0.13.2\n",
      "\n",
      "To fix this you could try to:\n",
      "1. loosen the range of package versions you've specified\n",
      "2. remove package versions to allow pip attempt to solve the dependency conflict\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\python.exe' 'C:\\Users\\shm22004\\AppData\\Local\\Temp\\pip-standalone-pip-5n7hx4px\\__env_pip__.zip\\pip' install --ignore-installed --no-user --prefix 'C:\\Users\\shm22004\\AppData\\Local\\Temp\\pip-build-env-n1ctlf5o\\overlay' --no-warn-script-location --no-binary :none: --only-binary :none: -i https://pypi.org/simple -- 'cython >= 0.29' 'numpy==1.14.5; python_version<'\"'\"'3.7'\"'\"'' 'numpy==1.16.0; python_version>='\"'\"'3.7'\"'\"'' setuptools setuptools_scm wheel\n",
      "       cwd: None\n",
      "  Complete output (586 lines):\n",
      "  Ignoring numpy: markers 'python_version < \"3.7\"' don't match your environment\n",
      "  Collecting cython>=0.29\n",
      "    Using cached Cython-0.29.24-cp39-cp39-win_amd64.whl (1.7 MB)\n",
      "  Collecting numpy==1.16.0\n",
      "    Using cached numpy-1.16.0.zip (5.1 MB)\n",
      "    Preparing metadata (setup.py): started\n",
      "    Preparing metadata (setup.py): finished with status 'done'\n",
      "  Collecting setuptools\n",
      "    Using cached setuptools-58.3.0-py3-none-any.whl (946 kB)\n",
      "  Collecting setuptools_scm\n",
      "    Using cached setuptools_scm-6.3.2-py3-none-any.whl (33 kB)\n",
      "  Collecting wheel\n",
      "    Using cached wheel-0.37.0-py2.py3-none-any.whl (35 kB)\n",
      "  Collecting tomli>=1.0.0\n",
      "    Using cached tomli-1.2.2-py3-none-any.whl (12 kB)\n",
      "  Collecting packaging>=20.0\n",
      "    Using cached packaging-21.2-py3-none-any.whl (40 kB)\n",
      "  Collecting pyparsing<3,>=2.0.2\n",
      "    Using cached pyparsing-2.4.7-py2.py3-none-any.whl (67 kB)\n",
      "  Building wheels for collected packages: numpy\n",
      "    Building wheel for numpy (setup.py): started\n",
      "    Building wheel for numpy (setup.py): finished with status 'error'\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\shm22004\\\\AppData\\\\Local\\\\Temp\\\\pip-install-p_pszwjb\\\\numpy_dc577ece64264edaaf59e335b5e5b13a\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\shm22004\\\\AppData\\\\Local\\\\Temp\\\\pip-install-p_pszwjb\\\\numpy_dc577ece64264edaaf59e335b5e5b13a\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' bdist_wheel -d 'C:\\Users\\shm22004\\AppData\\Local\\Temp\\pip-wheel-ftza_ozc'\n",
      "         cwd: C:\\Users\\shm22004\\AppData\\Local\\Temp\\pip-install-p_pszwjb\\numpy_dc577ece64264edaaf59e335b5e5b13a\\\n",
      "    Complete output (264 lines):\n",
      "    Running from numpy source directory.\n",
      "    C:\\Users\\shm22004\\AppData\\Local\\Temp\\pip-install-p_pszwjb\\numpy_dc577ece64264edaaf59e335b5e5b13a\\numpy\\distutils\\misc_util.py:476: SyntaxWarning: \"is\" with a literal. Did you mean \"==\"?\n",
      "      return is_string(s) and ('*' in s or '?' is s)\n",
      "    blas_opt_info:\n",
      "    blas_mkl_info:\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries mkl_rt not found in ['C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\lib', 'C:\\\\', 'C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\libs']\n",
      "      NOT AVAILABLE\n",
      "  \n",
      "    blis_info:\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries blis not found in ['C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\lib', 'C:\\\\', 'C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\libs']\n",
      "      NOT AVAILABLE\n",
      "  \n",
      "    openblas_info:\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries openblas not found in ['C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\lib', 'C:\\\\', 'C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\libs']\n",
      "    get_default_fcompiler: matching types: '['gnu', 'intelv', 'absoft', 'compaqv', 'intelev', 'gnu95', 'g95', 'intelvem', 'intelem', 'flang']'\n",
      "    customize GnuFCompiler\n",
      "    Could not locate executable g77\n",
      "    Could not locate executable f77\n",
      "    customize IntelVisualFCompiler\n",
      "    Could not locate executable ifort\n",
      "    Could not locate executable ifl\n",
      "    customize AbsoftFCompiler\n",
      "    Could not locate executable f90\n",
      "    customize CompaqVisualFCompiler\n",
      "    Could not locate executable DF\n",
      "    customize IntelItaniumVisualFCompiler\n",
      "    Could not locate executable efl\n",
      "    customize Gnu95FCompiler\n",
      "    Could not locate executable gfortran\n",
      "    Could not locate executable f95\n",
      "    customize G95FCompiler\n",
      "    Could not locate executable g95\n",
      "    customize IntelEM64VisualFCompiler\n",
      "    customize IntelEM64TFCompiler\n",
      "    Could not locate executable efort\n",
      "    Could not locate executable efc\n",
      "    customize PGroupFlangCompiler\n",
      "    Could not locate executable flang\n",
      "    don't know how to compile Fortran code on platform 'nt'\n",
      "      NOT AVAILABLE\n",
      "  \n",
      "    atlas_3_10_blas_threads_info:\n",
      "    Setting PTATLAS=ATLAS\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries tatlas not found in ['C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\lib', 'C:\\\\', 'C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\libs']\n",
      "      NOT AVAILABLE\n",
      "  \n",
      "    atlas_3_10_blas_info:\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries satlas not found in ['C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\lib', 'C:\\\\', 'C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\libs']\n",
      "      NOT AVAILABLE\n",
      "  \n",
      "    atlas_blas_threads_info:\n",
      "    Setting PTATLAS=ATLAS\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries ptf77blas,ptcblas,atlas not found in ['C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\lib', 'C:\\\\', 'C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\libs']\n",
      "      NOT AVAILABLE\n",
      "  \n",
      "    atlas_blas_info:\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries f77blas,cblas,atlas not found in ['C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\lib', 'C:\\\\', 'C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\libs']\n",
      "      NOT AVAILABLE\n",
      "  \n",
      "    accelerate_info:\n",
      "      NOT AVAILABLE\n",
      "  \n",
      "    C:\\Users\\shm22004\\AppData\\Local\\Temp\\pip-install-p_pszwjb\\numpy_dc577ece64264edaaf59e335b5e5b13a\\numpy\\distutils\\system_info.py:625: UserWarning:\n",
      "        Atlas (http://math-atlas.sourceforge.net/) libraries not found.\n",
      "        Directories to search for the libraries can be specified in the\n",
      "        numpy/distutils/site.cfg file (section [atlas]) or by setting\n",
      "        the ATLAS environment variable.\n",
      "      self.calc_info()\n",
      "    blas_info:\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries blas not found in ['C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\lib', 'C:\\\\', 'C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\libs']\n",
      "      NOT AVAILABLE\n",
      "  \n",
      "    C:\\Users\\shm22004\\AppData\\Local\\Temp\\pip-install-p_pszwjb\\numpy_dc577ece64264edaaf59e335b5e5b13a\\numpy\\distutils\\system_info.py:625: UserWarning:\n",
      "        Blas (http://www.netlib.org/blas/) libraries not found.\n",
      "        Directories to search for the libraries can be specified in the\n",
      "        numpy/distutils/site.cfg file (section [blas]) or by setting\n",
      "        the BLAS environment variable.\n",
      "      self.calc_info()\n",
      "    blas_src_info:\n",
      "      NOT AVAILABLE\n",
      "  \n",
      "    C:\\Users\\shm22004\\AppData\\Local\\Temp\\pip-install-p_pszwjb\\numpy_dc577ece64264edaaf59e335b5e5b13a\\numpy\\distutils\\system_info.py:625: UserWarning:\n",
      "        Blas (http://www.netlib.org/blas/) sources not found.\n",
      "        Directories to search for the sources can be specified in the\n",
      "        numpy/distutils/site.cfg file (section [blas_src]) or by setting\n",
      "        the BLAS_SRC environment variable.\n",
      "      self.calc_info()\n",
      "      NOT AVAILABLE\n",
      "  \n",
      "    'svnversion' is not recognized as an internal or external command,\n",
      "    operable program or batch file.\n",
      "    non-existing path in 'numpy\\\\distutils': 'site.cfg'\n",
      "    lapack_opt_info:\n",
      "    lapack_mkl_info:\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries mkl_rt not found in ['C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\lib', 'C:\\\\', 'C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\libs']\n",
      "      NOT AVAILABLE\n",
      "  \n",
      "    openblas_lapack_info:\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries openblas not found in ['C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\lib', 'C:\\\\', 'C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\libs']\n",
      "      NOT AVAILABLE\n",
      "  \n",
      "    openblas_clapack_info:\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries openblas,lapack not found in ['C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\lib', 'C:\\\\', 'C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\libs']\n",
      "      NOT AVAILABLE\n",
      "  \n",
      "    atlas_3_10_threads_info:\n",
      "    Setting PTATLAS=ATLAS\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries lapack_atlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries tatlas,tatlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries lapack_atlas not found in C:\\\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries tatlas,tatlas not found in C:\\\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries lapack_atlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\libs\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries tatlas,tatlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\libs\n",
      "    <class 'numpy.distutils.system_info.atlas_3_10_threads_info'>\n",
      "      NOT AVAILABLE\n",
      "  \n",
      "    atlas_3_10_info:\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries lapack_atlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries satlas,satlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries lapack_atlas not found in C:\\\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries satlas,satlas not found in C:\\\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries lapack_atlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\libs\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries satlas,satlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\libs\n",
      "    <class 'numpy.distutils.system_info.atlas_3_10_info'>\n",
      "      NOT AVAILABLE\n",
      "  \n",
      "    atlas_threads_info:\n",
      "    Setting PTATLAS=ATLAS\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries lapack_atlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries ptf77blas,ptcblas,atlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries lapack_atlas not found in C:\\\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries ptf77blas,ptcblas,atlas not found in C:\\\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries lapack_atlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\libs\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries ptf77blas,ptcblas,atlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\libs\n",
      "    <class 'numpy.distutils.system_info.atlas_threads_info'>\n",
      "      NOT AVAILABLE\n",
      "  \n",
      "    atlas_info:\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries lapack_atlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries f77blas,cblas,atlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries lapack_atlas not found in C:\\\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries f77blas,cblas,atlas not found in C:\\\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries lapack_atlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\libs\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries f77blas,cblas,atlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\libs\n",
      "    <class 'numpy.distutils.system_info.atlas_info'>\n",
      "      NOT AVAILABLE\n",
      "  \n",
      "    lapack_info:\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries lapack not found in ['C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\lib', 'C:\\\\', 'C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\libs']\n",
      "      NOT AVAILABLE\n",
      "  \n",
      "    C:\\Users\\shm22004\\AppData\\Local\\Temp\\pip-install-p_pszwjb\\numpy_dc577ece64264edaaf59e335b5e5b13a\\numpy\\distutils\\system_info.py:625: UserWarning:\n",
      "        Lapack (http://www.netlib.org/lapack/) libraries not found.\n",
      "        Directories to search for the libraries can be specified in the\n",
      "        numpy/distutils/site.cfg file (section [lapack]) or by setting\n",
      "        the LAPACK environment variable.\n",
      "      self.calc_info()\n",
      "    lapack_src_info:\n",
      "      NOT AVAILABLE\n",
      "  \n",
      "    C:\\Users\\shm22004\\AppData\\Local\\Temp\\pip-install-p_pszwjb\\numpy_dc577ece64264edaaf59e335b5e5b13a\\numpy\\distutils\\system_info.py:625: UserWarning:\n",
      "        Lapack (http://www.netlib.org/lapack/) sources not found.\n",
      "        Directories to search for the sources can be specified in the\n",
      "        numpy/distutils/site.cfg file (section [lapack_src]) or by setting\n",
      "        the LAPACK_SRC environment variable.\n",
      "      self.calc_info()\n",
      "      NOT AVAILABLE\n",
      "  \n",
      "    C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\\distutils\\dist.py:274: UserWarning: Unknown distribution option: 'define_macros'\n",
      "      warnings.warn(msg)\n",
      "    running bdist_wheel\n",
      "    running build\n",
      "    running config_cc\n",
      "    unifing config_cc, config, build_clib, build_ext, build commands --compiler options\n",
      "    running config_fc\n",
      "    unifing config_fc, config, build_clib, build_ext, build commands --fcompiler options\n",
      "    running build_src\n",
      "    build_src\n",
      "    building py_modules sources\n",
      "    creating build\n",
      "    creating build\\src.win-amd64-3.9\n",
      "    creating build\\src.win-amd64-3.9\\numpy\n",
      "    creating build\\src.win-amd64-3.9\\numpy\\distutils\n",
      "    building library \"npymath\" sources\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    error: Microsoft Visual C++ 14.0 or greater is required. Get it with \"Microsoft C++ Build Tools\": https://visualstudio.microsoft.com/visual-cpp-build-tools/\n",
      "    ----------------------------------------\n",
      "    ERROR: Failed building wheel for numpy\n",
      "    Running setup.py clean for numpy\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\shm22004\\\\AppData\\\\Local\\\\Temp\\\\pip-install-p_pszwjb\\\\numpy_dc577ece64264edaaf59e335b5e5b13a\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\shm22004\\\\AppData\\\\Local\\\\Temp\\\\pip-install-p_pszwjb\\\\numpy_dc577ece64264edaaf59e335b5e5b13a\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' clean --all\n",
      "         cwd: C:\\Users\\shm22004\\AppData\\Local\\Temp\\pip-install-p_pszwjb\\numpy_dc577ece64264edaaf59e335b5e5b13a\n",
      "    Complete output (10 lines):\n",
      "    Running from numpy source directory.\n",
      "  \n",
      "    `setup.py clean` is not supported, use one of the following instead:\n",
      "  \n",
      "      - `git clean -xdf` (cleans all files)\n",
      "      - `git clean -Xdf` (cleans all versioned files, doesn't touch\n",
      "                          files that aren't checked into the git repo)\n",
      "  \n",
      "    Add `--force` to your command to use it anyway if you must (unsupported).\n",
      "  \n",
      "    ----------------------------------------\n",
      "    ERROR: Failed cleaning build dir for numpy\n",
      "  Failed to build numpy\n",
      "  Installing collected packages: pyparsing, tomli, setuptools, packaging, wheel, setuptools-scm, numpy, cython\n",
      "      Running setup.py install for numpy: started\n",
      "      Running setup.py install for numpy: finished with status 'error'\n",
      "      ERROR: Command errored out with exit status 1:\n",
      "       command: 'C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\shm22004\\\\AppData\\\\Local\\\\Temp\\\\pip-install-p_pszwjb\\\\numpy_dc577ece64264edaaf59e335b5e5b13a\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\shm22004\\\\AppData\\\\Local\\\\Temp\\\\pip-install-p_pszwjb\\\\numpy_dc577ece64264edaaf59e335b5e5b13a\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' install --record 'C:\\Users\\shm22004\\AppData\\Local\\Temp\\pip-record-mw18fjts\\install-record.txt' --single-version-externally-managed --prefix 'C:\\Users\\shm22004\\AppData\\Local\\Temp\\pip-build-env-n1ctlf5o\\overlay' --compile --install-headers 'C:\\Users\\shm22004\\AppData\\Local\\Temp\\pip-build-env-n1ctlf5o\\overlay\\Include\\numpy'\n",
      "           cwd: C:\\Users\\shm22004\\AppData\\Local\\Temp\\pip-install-p_pszwjb\\numpy_dc577ece64264edaaf59e335b5e5b13a\\\n",
      "      Complete output (267 lines):\n",
      "      Running from numpy source directory.\n",
      "  \n",
      "      Note: if you need reliable uninstall behavior, then install\n",
      "      with pip instead of using `setup.py install`:\n",
      "  \n",
      "        - `pip install .`       (from a git repo or downloaded source\n",
      "                                 release)\n",
      "        - `pip install numpy`   (last NumPy release on PyPi)\n",
      "  \n",
      "  \n",
      "      blas_opt_info:\n",
      "      blas_mkl_info:\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries mkl_rt not found in ['C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\lib', 'C:\\\\', 'C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\libs']\n",
      "        NOT AVAILABLE\n",
      "  \n",
      "      blis_info:\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries blis not found in ['C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\lib', 'C:\\\\', 'C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\libs']\n",
      "        NOT AVAILABLE\n",
      "  \n",
      "      openblas_info:\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries openblas not found in ['C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\lib', 'C:\\\\', 'C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\libs']\n",
      "      get_default_fcompiler: matching types: '['gnu', 'intelv', 'absoft', 'compaqv', 'intelev', 'gnu95', 'g95', 'intelvem', 'intelem', 'flang']'\n",
      "      customize GnuFCompiler\n",
      "      Could not locate executable g77\n",
      "      Could not locate executable f77\n",
      "      customize IntelVisualFCompiler\n",
      "      Could not locate executable ifort\n",
      "      Could not locate executable ifl\n",
      "      customize AbsoftFCompiler\n",
      "      Could not locate executable f90\n",
      "      customize CompaqVisualFCompiler\n",
      "      Could not locate executable DF\n",
      "      customize IntelItaniumVisualFCompiler\n",
      "      Could not locate executable efl\n",
      "      customize Gnu95FCompiler\n",
      "      Could not locate executable gfortran\n",
      "      Could not locate executable f95\n",
      "      customize G95FCompiler\n",
      "      Could not locate executable g95\n",
      "      customize IntelEM64VisualFCompiler\n",
      "      customize IntelEM64TFCompiler\n",
      "      Could not locate executable efort\n",
      "      Could not locate executable efc\n",
      "      customize PGroupFlangCompiler\n",
      "      Could not locate executable flang\n",
      "      don't know how to compile Fortran code on platform 'nt'\n",
      "        NOT AVAILABLE\n",
      "  \n",
      "      atlas_3_10_blas_threads_info:\n",
      "      Setting PTATLAS=ATLAS\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries tatlas not found in ['C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\lib', 'C:\\\\', 'C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\libs']\n",
      "        NOT AVAILABLE\n",
      "  \n",
      "      atlas_3_10_blas_info:\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries satlas not found in ['C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\lib', 'C:\\\\', 'C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\libs']\n",
      "        NOT AVAILABLE\n",
      "  \n",
      "      atlas_blas_threads_info:\n",
      "      Setting PTATLAS=ATLAS\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries ptf77blas,ptcblas,atlas not found in ['C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\lib', 'C:\\\\', 'C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\libs']\n",
      "        NOT AVAILABLE\n",
      "  \n",
      "      atlas_blas_info:\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries f77blas,cblas,atlas not found in ['C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\lib', 'C:\\\\', 'C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\libs']\n",
      "        NOT AVAILABLE\n",
      "  \n",
      "      accelerate_info:\n",
      "        NOT AVAILABLE\n",
      "  \n",
      "      C:\\Users\\shm22004\\AppData\\Local\\Temp\\pip-install-p_pszwjb\\numpy_dc577ece64264edaaf59e335b5e5b13a\\numpy\\distutils\\system_info.py:625: UserWarning:\n",
      "          Atlas (http://math-atlas.sourceforge.net/) libraries not found.\n",
      "          Directories to search for the libraries can be specified in the\n",
      "          numpy/distutils/site.cfg file (section [atlas]) or by setting\n",
      "          the ATLAS environment variable.\n",
      "        self.calc_info()\n",
      "      blas_info:\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries blas not found in ['C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\lib', 'C:\\\\', 'C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\libs']\n",
      "        NOT AVAILABLE\n",
      "  \n",
      "      C:\\Users\\shm22004\\AppData\\Local\\Temp\\pip-install-p_pszwjb\\numpy_dc577ece64264edaaf59e335b5e5b13a\\numpy\\distutils\\system_info.py:625: UserWarning:\n",
      "          Blas (http://www.netlib.org/blas/) libraries not found.\n",
      "          Directories to search for the libraries can be specified in the\n",
      "          numpy/distutils/site.cfg file (section [blas]) or by setting\n",
      "          the BLAS environment variable.\n",
      "        self.calc_info()\n",
      "      blas_src_info:\n",
      "        NOT AVAILABLE\n",
      "  \n",
      "      C:\\Users\\shm22004\\AppData\\Local\\Temp\\pip-install-p_pszwjb\\numpy_dc577ece64264edaaf59e335b5e5b13a\\numpy\\distutils\\system_info.py:625: UserWarning:\n",
      "          Blas (http://www.netlib.org/blas/) sources not found.\n",
      "          Directories to search for the sources can be specified in the\n",
      "          numpy/distutils/site.cfg file (section [blas_src]) or by setting\n",
      "          the BLAS_SRC environment variable.\n",
      "        self.calc_info()\n",
      "        NOT AVAILABLE\n",
      "  \n",
      "      'svnversion' is not recognized as an internal or external command,\n",
      "      operable program or batch file.\n",
      "      non-existing path in 'numpy\\\\distutils': 'site.cfg'\n",
      "      lapack_opt_info:\n",
      "      lapack_mkl_info:\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries mkl_rt not found in ['C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\lib', 'C:\\\\', 'C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\libs']\n",
      "        NOT AVAILABLE\n",
      "  \n",
      "      openblas_lapack_info:\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries openblas not found in ['C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\lib', 'C:\\\\', 'C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\libs']\n",
      "        NOT AVAILABLE\n",
      "  \n",
      "      openblas_clapack_info:\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries openblas,lapack not found in ['C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\lib', 'C:\\\\', 'C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\libs']\n",
      "        NOT AVAILABLE\n",
      "  \n",
      "      atlas_3_10_threads_info:\n",
      "      Setting PTATLAS=ATLAS\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries lapack_atlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries tatlas,tatlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries lapack_atlas not found in C:\\\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries tatlas,tatlas not found in C:\\\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries lapack_atlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\libs\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries tatlas,tatlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\libs\n",
      "      <class 'numpy.distutils.system_info.atlas_3_10_threads_info'>\n",
      "        NOT AVAILABLE\n",
      "  \n",
      "      atlas_3_10_info:\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries lapack_atlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries satlas,satlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries lapack_atlas not found in C:\\\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries satlas,satlas not found in C:\\\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries lapack_atlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\libs\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries satlas,satlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\libs\n",
      "      <class 'numpy.distutils.system_info.atlas_3_10_info'>\n",
      "        NOT AVAILABLE\n",
      "  \n",
      "      atlas_threads_info:\n",
      "      Setting PTATLAS=ATLAS\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries lapack_atlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries ptf77blas,ptcblas,atlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries lapack_atlas not found in C:\\\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries ptf77blas,ptcblas,atlas not found in C:\\\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries lapack_atlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\libs\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries ptf77blas,ptcblas,atlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\libs\n",
      "      <class 'numpy.distutils.system_info.atlas_threads_info'>\n",
      "        NOT AVAILABLE\n",
      "  \n",
      "      atlas_info:\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries lapack_atlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries f77blas,cblas,atlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries lapack_atlas not found in C:\\\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries f77blas,cblas,atlas not found in C:\\\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries lapack_atlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\libs\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries f77blas,cblas,atlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\libs\n",
      "      <class 'numpy.distutils.system_info.atlas_info'>\n",
      "        NOT AVAILABLE\n",
      "  \n",
      "      lapack_info:\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries lapack not found in ['C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\lib', 'C:\\\\', 'C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\libs']\n",
      "        NOT AVAILABLE\n",
      "  \n",
      "      C:\\Users\\shm22004\\AppData\\Local\\Temp\\pip-install-p_pszwjb\\numpy_dc577ece64264edaaf59e335b5e5b13a\\numpy\\distutils\\system_info.py:625: UserWarning:\n",
      "          Lapack (http://www.netlib.org/lapack/) libraries not found.\n",
      "          Directories to search for the libraries can be specified in the\n",
      "          numpy/distutils/site.cfg file (section [lapack]) or by setting\n",
      "          the LAPACK environment variable.\n",
      "        self.calc_info()\n",
      "      lapack_src_info:\n",
      "        NOT AVAILABLE\n",
      "  \n",
      "      C:\\Users\\shm22004\\AppData\\Local\\Temp\\pip-install-p_pszwjb\\numpy_dc577ece64264edaaf59e335b5e5b13a\\numpy\\distutils\\system_info.py:625: UserWarning:\n",
      "          Lapack (http://www.netlib.org/lapack/) sources not found.\n",
      "          Directories to search for the sources can be specified in the\n",
      "          numpy/distutils/site.cfg file (section [lapack_src]) or by setting\n",
      "          the LAPACK_SRC environment variable.\n",
      "        self.calc_info()\n",
      "        NOT AVAILABLE\n",
      "  \n",
      "      C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\\distutils\\dist.py:274: UserWarning: Unknown distribution option: 'define_macros'\n",
      "        warnings.warn(msg)\n",
      "      running install\n",
      "      running build\n",
      "      running config_cc\n",
      "      unifing config_cc, config, build_clib, build_ext, build commands --compiler options\n",
      "      running config_fc\n",
      "      unifing config_fc, config, build_clib, build_ext, build commands --fcompiler options\n",
      "      running build_src\n",
      "      build_src\n",
      "      building py_modules sources\n",
      "      building library \"npymath\" sources\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      error: Microsoft Visual C++ 14.0 or greater is required. Get it with \"Microsoft C++ Build Tools\": https://visualstudio.microsoft.com/visual-cpp-build-tools/\n",
      "      ----------------------------------------\n",
      "  ERROR: Command errored out with exit status 1: 'C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\shm22004\\\\AppData\\\\Local\\\\Temp\\\\pip-install-p_pszwjb\\\\numpy_dc577ece64264edaaf59e335b5e5b13a\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\shm22004\\\\AppData\\\\Local\\\\Temp\\\\pip-install-p_pszwjb\\\\numpy_dc577ece64264edaaf59e335b5e5b13a\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' install --record 'C:\\Users\\shm22004\\AppData\\Local\\Temp\\pip-record-mw18fjts\\install-record.txt' --single-version-externally-managed --prefix 'C:\\Users\\shm22004\\AppData\\Local\\Temp\\pip-build-env-n1ctlf5o\\overlay' --compile --install-headers 'C:\\Users\\shm22004\\AppData\\Local\\Temp\\pip-build-env-n1ctlf5o\\overlay\\Include\\numpy' Check the logs for full command output.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/ed/c9/85d179d5a0575e1b066fb94bfe1e37b6d3ca546b58e75b9d1ca4952320de/pyarrow-0.17.1.tar.gz#sha256=278d11800c2e0f9bea6314ef718b2368b4046ba24b6c631c14edad5a1d351e49 (from https://pypi.org/simple/pyarrow/) (requires-python:>=3.5). Command errored out with exit status 1: 'C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\python.exe' 'C:\\Users\\shm22004\\AppData\\Local\\Temp\\pip-standalone-pip-5n7hx4px\\__env_pip__.zip\\pip' install --ignore-installed --no-user --prefix 'C:\\Users\\shm22004\\AppData\\Local\\Temp\\pip-build-env-n1ctlf5o\\overlay' --no-warn-script-location --no-binary :none: --only-binary :none: -i https://pypi.org/simple -- 'cython >= 0.29' 'numpy==1.14.5; python_version<'\"'\"'3.7'\"'\"'' 'numpy==1.16.0; python_version>='\"'\"'3.7'\"'\"'' setuptools setuptools_scm wheel Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\python.exe' 'C:\\Users\\shm22004\\AppData\\Local\\Temp\\pip-standalone-pip-h7q1_5fr\\__env_pip__.zip\\pip' install --ignore-installed --no-user --prefix 'C:\\Users\\shm22004\\AppData\\Local\\Temp\\pip-build-env-s8hp1rbk\\overlay' --no-warn-script-location --no-binary :none: --only-binary :none: -i https://pypi.org/simple -- 'cython >= 0.29' 'numpy==1.14.5; python_version<'\"'\"'3.7'\"'\"'' 'numpy==1.16.0; python_version>='\"'\"'3.7'\"'\"'' setuptools setuptools_scm wheel\n",
      "       cwd: None\n",
      "  Complete output (586 lines):\n",
      "  Ignoring numpy: markers 'python_version < \"3.7\"' don't match your environment\n",
      "  Collecting cython>=0.29\n",
      "    Using cached Cython-0.29.24-cp39-cp39-win_amd64.whl (1.7 MB)\n",
      "  Collecting numpy==1.16.0\n",
      "    Using cached numpy-1.16.0.zip (5.1 MB)\n",
      "    Preparing metadata (setup.py): started\n",
      "    Preparing metadata (setup.py): finished with status 'done'\n",
      "  Collecting setuptools\n",
      "    Using cached setuptools-58.3.0-py3-none-any.whl (946 kB)\n",
      "  Collecting setuptools_scm\n",
      "    Using cached setuptools_scm-6.3.2-py3-none-any.whl (33 kB)\n",
      "  Collecting wheel\n",
      "    Using cached wheel-0.37.0-py2.py3-none-any.whl (35 kB)\n",
      "  Collecting packaging>=20.0\n",
      "    Using cached packaging-21.2-py3-none-any.whl (40 kB)\n",
      "  Collecting tomli>=1.0.0\n",
      "    Using cached tomli-1.2.2-py3-none-any.whl (12 kB)\n",
      "  Collecting pyparsing<3,>=2.0.2\n",
      "    Using cached pyparsing-2.4.7-py2.py3-none-any.whl (67 kB)\n",
      "  Building wheels for collected packages: numpy\n",
      "    Building wheel for numpy (setup.py): started\n",
      "    Building wheel for numpy (setup.py): finished with status 'error'\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\shm22004\\\\AppData\\\\Local\\\\Temp\\\\pip-install-n2b5b22y\\\\numpy_a28c2754541940f18367734d5cdccbc7\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\shm22004\\\\AppData\\\\Local\\\\Temp\\\\pip-install-n2b5b22y\\\\numpy_a28c2754541940f18367734d5cdccbc7\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' bdist_wheel -d 'C:\\Users\\shm22004\\AppData\\Local\\Temp\\pip-wheel-ykke1zyr'\n",
      "         cwd: C:\\Users\\shm22004\\AppData\\Local\\Temp\\pip-install-n2b5b22y\\numpy_a28c2754541940f18367734d5cdccbc7\\\n",
      "    Complete output (264 lines):\n",
      "    Running from numpy source directory.\n",
      "    C:\\Users\\shm22004\\AppData\\Local\\Temp\\pip-install-n2b5b22y\\numpy_a28c2754541940f18367734d5cdccbc7\\numpy\\distutils\\misc_util.py:476: SyntaxWarning: \"is\" with a literal. Did you mean \"==\"?\n",
      "      return is_string(s) and ('*' in s or '?' is s)\n",
      "    blas_opt_info:\n",
      "    blas_mkl_info:\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries mkl_rt not found in ['C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\lib', 'C:\\\\', 'C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\libs']\n",
      "      NOT AVAILABLE\n",
      "  \n",
      "    blis_info:\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries blis not found in ['C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\lib', 'C:\\\\', 'C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\libs']\n",
      "      NOT AVAILABLE\n",
      "  \n",
      "    openblas_info:\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries openblas not found in ['C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\lib', 'C:\\\\', 'C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\libs']\n",
      "    get_default_fcompiler: matching types: '['gnu', 'intelv', 'absoft', 'compaqv', 'intelev', 'gnu95', 'g95', 'intelvem', 'intelem', 'flang']'\n",
      "    customize GnuFCompiler\n",
      "    Could not locate executable g77\n",
      "    Could not locate executable f77\n",
      "    customize IntelVisualFCompiler\n",
      "    Could not locate executable ifort\n",
      "    Could not locate executable ifl\n",
      "    customize AbsoftFCompiler\n",
      "    Could not locate executable f90\n",
      "    customize CompaqVisualFCompiler\n",
      "    Could not locate executable DF\n",
      "    customize IntelItaniumVisualFCompiler\n",
      "    Could not locate executable efl\n",
      "    customize Gnu95FCompiler\n",
      "    Could not locate executable gfortran\n",
      "    Could not locate executable f95\n",
      "    customize G95FCompiler\n",
      "    Could not locate executable g95\n",
      "    customize IntelEM64VisualFCompiler\n",
      "    customize IntelEM64TFCompiler\n",
      "    Could not locate executable efort\n",
      "    Could not locate executable efc\n",
      "    customize PGroupFlangCompiler\n",
      "    Could not locate executable flang\n",
      "    don't know how to compile Fortran code on platform 'nt'\n",
      "      NOT AVAILABLE\n",
      "  \n",
      "    atlas_3_10_blas_threads_info:\n",
      "    Setting PTATLAS=ATLAS\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries tatlas not found in ['C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\lib', 'C:\\\\', 'C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\libs']\n",
      "      NOT AVAILABLE\n",
      "  \n",
      "    atlas_3_10_blas_info:\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries satlas not found in ['C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\lib', 'C:\\\\', 'C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\libs']\n",
      "      NOT AVAILABLE\n",
      "  \n",
      "    atlas_blas_threads_info:\n",
      "    Setting PTATLAS=ATLAS\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries ptf77blas,ptcblas,atlas not found in ['C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\lib', 'C:\\\\', 'C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\libs']\n",
      "      NOT AVAILABLE\n",
      "  \n",
      "    atlas_blas_info:\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries f77blas,cblas,atlas not found in ['C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\lib', 'C:\\\\', 'C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\libs']\n",
      "      NOT AVAILABLE\n",
      "  \n",
      "    accelerate_info:\n",
      "      NOT AVAILABLE\n",
      "  \n",
      "    C:\\Users\\shm22004\\AppData\\Local\\Temp\\pip-install-n2b5b22y\\numpy_a28c2754541940f18367734d5cdccbc7\\numpy\\distutils\\system_info.py:625: UserWarning:\n",
      "        Atlas (http://math-atlas.sourceforge.net/) libraries not found.\n",
      "        Directories to search for the libraries can be specified in the\n",
      "        numpy/distutils/site.cfg file (section [atlas]) or by setting\n",
      "        the ATLAS environment variable.\n",
      "      self.calc_info()\n",
      "    blas_info:\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries blas not found in ['C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\lib', 'C:\\\\', 'C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\libs']\n",
      "      NOT AVAILABLE\n",
      "  \n",
      "    C:\\Users\\shm22004\\AppData\\Local\\Temp\\pip-install-n2b5b22y\\numpy_a28c2754541940f18367734d5cdccbc7\\numpy\\distutils\\system_info.py:625: UserWarning:\n",
      "        Blas (http://www.netlib.org/blas/) libraries not found.\n",
      "        Directories to search for the libraries can be specified in the\n",
      "        numpy/distutils/site.cfg file (section [blas]) or by setting\n",
      "        the BLAS environment variable.\n",
      "      self.calc_info()\n",
      "    blas_src_info:\n",
      "      NOT AVAILABLE\n",
      "  \n",
      "    C:\\Users\\shm22004\\AppData\\Local\\Temp\\pip-install-n2b5b22y\\numpy_a28c2754541940f18367734d5cdccbc7\\numpy\\distutils\\system_info.py:625: UserWarning:\n",
      "        Blas (http://www.netlib.org/blas/) sources not found.\n",
      "        Directories to search for the sources can be specified in the\n",
      "        numpy/distutils/site.cfg file (section [blas_src]) or by setting\n",
      "        the BLAS_SRC environment variable.\n",
      "      self.calc_info()\n",
      "      NOT AVAILABLE\n",
      "  \n",
      "    'svnversion' is not recognized as an internal or external command,\n",
      "    operable program or batch file.\n",
      "    non-existing path in 'numpy\\\\distutils': 'site.cfg'\n",
      "    lapack_opt_info:\n",
      "    lapack_mkl_info:\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries mkl_rt not found in ['C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\lib', 'C:\\\\', 'C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\libs']\n",
      "      NOT AVAILABLE\n",
      "  \n",
      "    openblas_lapack_info:\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries openblas not found in ['C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\lib', 'C:\\\\', 'C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\libs']\n",
      "      NOT AVAILABLE\n",
      "  \n",
      "    openblas_clapack_info:\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries openblas,lapack not found in ['C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\lib', 'C:\\\\', 'C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\libs']\n",
      "      NOT AVAILABLE\n",
      "  \n",
      "    atlas_3_10_threads_info:\n",
      "    Setting PTATLAS=ATLAS\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries lapack_atlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries tatlas,tatlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries lapack_atlas not found in C:\\\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries tatlas,tatlas not found in C:\\\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries lapack_atlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\libs\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries tatlas,tatlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\libs\n",
      "    <class 'numpy.distutils.system_info.atlas_3_10_threads_info'>\n",
      "      NOT AVAILABLE\n",
      "  \n",
      "    atlas_3_10_info:\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries lapack_atlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries satlas,satlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries lapack_atlas not found in C:\\\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries satlas,satlas not found in C:\\\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries lapack_atlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\libs\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries satlas,satlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\libs\n",
      "    <class 'numpy.distutils.system_info.atlas_3_10_info'>\n",
      "      NOT AVAILABLE\n",
      "  \n",
      "    atlas_threads_info:\n",
      "    Setting PTATLAS=ATLAS\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries lapack_atlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries ptf77blas,ptcblas,atlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries lapack_atlas not found in C:\\\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries ptf77blas,ptcblas,atlas not found in C:\\\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries lapack_atlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\libs\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries ptf77blas,ptcblas,atlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\libs\n",
      "    <class 'numpy.distutils.system_info.atlas_threads_info'>\n",
      "      NOT AVAILABLE\n",
      "  \n",
      "    atlas_info:\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries lapack_atlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries f77blas,cblas,atlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries lapack_atlas not found in C:\\\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries f77blas,cblas,atlas not found in C:\\\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries lapack_atlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\libs\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries f77blas,cblas,atlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\libs\n",
      "    <class 'numpy.distutils.system_info.atlas_info'>\n",
      "      NOT AVAILABLE\n",
      "  \n",
      "    lapack_info:\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries lapack not found in ['C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\lib', 'C:\\\\', 'C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\libs']\n",
      "      NOT AVAILABLE\n",
      "  \n",
      "    C:\\Users\\shm22004\\AppData\\Local\\Temp\\pip-install-n2b5b22y\\numpy_a28c2754541940f18367734d5cdccbc7\\numpy\\distutils\\system_info.py:625: UserWarning:\n",
      "        Lapack (http://www.netlib.org/lapack/) libraries not found.\n",
      "        Directories to search for the libraries can be specified in the\n",
      "        numpy/distutils/site.cfg file (section [lapack]) or by setting\n",
      "        the LAPACK environment variable.\n",
      "      self.calc_info()\n",
      "    lapack_src_info:\n",
      "      NOT AVAILABLE\n",
      "  \n",
      "    C:\\Users\\shm22004\\AppData\\Local\\Temp\\pip-install-n2b5b22y\\numpy_a28c2754541940f18367734d5cdccbc7\\numpy\\distutils\\system_info.py:625: UserWarning:\n",
      "        Lapack (http://www.netlib.org/lapack/) sources not found.\n",
      "        Directories to search for the sources can be specified in the\n",
      "        numpy/distutils/site.cfg file (section [lapack_src]) or by setting\n",
      "        the LAPACK_SRC environment variable.\n",
      "      self.calc_info()\n",
      "      NOT AVAILABLE\n",
      "  \n",
      "    C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\\distutils\\dist.py:274: UserWarning: Unknown distribution option: 'define_macros'\n",
      "      warnings.warn(msg)\n",
      "    running bdist_wheel\n",
      "    running build\n",
      "    running config_cc\n",
      "    unifing config_cc, config, build_clib, build_ext, build commands --compiler options\n",
      "    running config_fc\n",
      "    unifing config_fc, config, build_clib, build_ext, build commands --fcompiler options\n",
      "    running build_src\n",
      "    build_src\n",
      "    building py_modules sources\n",
      "    creating build\n",
      "    creating build\\src.win-amd64-3.9\n",
      "    creating build\\src.win-amd64-3.9\\numpy\n",
      "    creating build\\src.win-amd64-3.9\\numpy\\distutils\n",
      "    building library \"npymath\" sources\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    error: Microsoft Visual C++ 14.0 or greater is required. Get it with \"Microsoft C++ Build Tools\": https://visualstudio.microsoft.com/visual-cpp-build-tools/\n",
      "    ----------------------------------------\n",
      "    ERROR: Failed building wheel for numpy\n",
      "    Running setup.py clean for numpy\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\shm22004\\\\AppData\\\\Local\\\\Temp\\\\pip-install-n2b5b22y\\\\numpy_a28c2754541940f18367734d5cdccbc7\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\shm22004\\\\AppData\\\\Local\\\\Temp\\\\pip-install-n2b5b22y\\\\numpy_a28c2754541940f18367734d5cdccbc7\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' clean --all\n",
      "         cwd: C:\\Users\\shm22004\\AppData\\Local\\Temp\\pip-install-n2b5b22y\\numpy_a28c2754541940f18367734d5cdccbc7\n",
      "    Complete output (10 lines):\n",
      "    Running from numpy source directory.\n",
      "  \n",
      "    `setup.py clean` is not supported, use one of the following instead:\n",
      "  \n",
      "      - `git clean -xdf` (cleans all files)\n",
      "      - `git clean -Xdf` (cleans all versioned files, doesn't touch\n",
      "                          files that aren't checked into the git repo)\n",
      "  \n",
      "    Add `--force` to your command to use it anyway if you must (unsupported).\n",
      "  \n",
      "    ----------------------------------------\n",
      "    ERROR: Failed cleaning build dir for numpy\n",
      "  Failed to build numpy\n",
      "  Installing collected packages: pyparsing, tomli, setuptools, packaging, wheel, setuptools-scm, numpy, cython\n",
      "      Running setup.py install for numpy: started\n",
      "      Running setup.py install for numpy: finished with status 'error'\n",
      "      ERROR: Command errored out with exit status 1:\n",
      "       command: 'C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\shm22004\\\\AppData\\\\Local\\\\Temp\\\\pip-install-n2b5b22y\\\\numpy_a28c2754541940f18367734d5cdccbc7\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\shm22004\\\\AppData\\\\Local\\\\Temp\\\\pip-install-n2b5b22y\\\\numpy_a28c2754541940f18367734d5cdccbc7\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' install --record 'C:\\Users\\shm22004\\AppData\\Local\\Temp\\pip-record-ba051w6x\\install-record.txt' --single-version-externally-managed --prefix 'C:\\Users\\shm22004\\AppData\\Local\\Temp\\pip-build-env-s8hp1rbk\\overlay' --compile --install-headers 'C:\\Users\\shm22004\\AppData\\Local\\Temp\\pip-build-env-s8hp1rbk\\overlay\\Include\\numpy'\n",
      "           cwd: C:\\Users\\shm22004\\AppData\\Local\\Temp\\pip-install-n2b5b22y\\numpy_a28c2754541940f18367734d5cdccbc7\\\n",
      "      Complete output (267 lines):\n",
      "      Running from numpy source directory.\n",
      "  \n",
      "      Note: if you need reliable uninstall behavior, then install\n",
      "      with pip instead of using `setup.py install`:\n",
      "  \n",
      "        - `pip install .`       (from a git repo or downloaded source\n",
      "                                 release)\n",
      "        - `pip install numpy`   (last NumPy release on PyPi)\n",
      "  \n",
      "  \n",
      "      blas_opt_info:\n",
      "      blas_mkl_info:\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries mkl_rt not found in ['C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\lib', 'C:\\\\', 'C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\libs']\n",
      "        NOT AVAILABLE\n",
      "  \n",
      "      blis_info:\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries blis not found in ['C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\lib', 'C:\\\\', 'C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\libs']\n",
      "        NOT AVAILABLE\n",
      "  \n",
      "      openblas_info:\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries openblas not found in ['C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\lib', 'C:\\\\', 'C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\libs']\n",
      "      get_default_fcompiler: matching types: '['gnu', 'intelv', 'absoft', 'compaqv', 'intelev', 'gnu95', 'g95', 'intelvem', 'intelem', 'flang']'\n",
      "      customize GnuFCompiler\n",
      "      Could not locate executable g77\n",
      "      Could not locate executable f77\n",
      "      customize IntelVisualFCompiler\n",
      "      Could not locate executable ifort\n",
      "      Could not locate executable ifl\n",
      "      customize AbsoftFCompiler\n",
      "      Could not locate executable f90\n",
      "      customize CompaqVisualFCompiler\n",
      "      Could not locate executable DF\n",
      "      customize IntelItaniumVisualFCompiler\n",
      "      Could not locate executable efl\n",
      "      customize Gnu95FCompiler\n",
      "      Could not locate executable gfortran\n",
      "      Could not locate executable f95\n",
      "      customize G95FCompiler\n",
      "      Could not locate executable g95\n",
      "      customize IntelEM64VisualFCompiler\n",
      "      customize IntelEM64TFCompiler\n",
      "      Could not locate executable efort\n",
      "      Could not locate executable efc\n",
      "      customize PGroupFlangCompiler\n",
      "      Could not locate executable flang\n",
      "      don't know how to compile Fortran code on platform 'nt'\n",
      "        NOT AVAILABLE\n",
      "  \n",
      "      atlas_3_10_blas_threads_info:\n",
      "      Setting PTATLAS=ATLAS\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries tatlas not found in ['C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\lib', 'C:\\\\', 'C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\libs']\n",
      "        NOT AVAILABLE\n",
      "  \n",
      "      atlas_3_10_blas_info:\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries satlas not found in ['C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\lib', 'C:\\\\', 'C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\libs']\n",
      "        NOT AVAILABLE\n",
      "  \n",
      "      atlas_blas_threads_info:\n",
      "      Setting PTATLAS=ATLAS\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries ptf77blas,ptcblas,atlas not found in ['C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\lib', 'C:\\\\', 'C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\libs']\n",
      "        NOT AVAILABLE\n",
      "  \n",
      "      atlas_blas_info:\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries f77blas,cblas,atlas not found in ['C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\lib', 'C:\\\\', 'C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\libs']\n",
      "        NOT AVAILABLE\n",
      "  \n",
      "      accelerate_info:\n",
      "        NOT AVAILABLE\n",
      "  \n",
      "      C:\\Users\\shm22004\\AppData\\Local\\Temp\\pip-install-n2b5b22y\\numpy_a28c2754541940f18367734d5cdccbc7\\numpy\\distutils\\system_info.py:625: UserWarning:\n",
      "          Atlas (http://math-atlas.sourceforge.net/) libraries not found.\n",
      "          Directories to search for the libraries can be specified in the\n",
      "          numpy/distutils/site.cfg file (section [atlas]) or by setting\n",
      "          the ATLAS environment variable.\n",
      "        self.calc_info()\n",
      "      blas_info:\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries blas not found in ['C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\lib', 'C:\\\\', 'C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\libs']\n",
      "        NOT AVAILABLE\n",
      "  \n",
      "      C:\\Users\\shm22004\\AppData\\Local\\Temp\\pip-install-n2b5b22y\\numpy_a28c2754541940f18367734d5cdccbc7\\numpy\\distutils\\system_info.py:625: UserWarning:\n",
      "          Blas (http://www.netlib.org/blas/) libraries not found.\n",
      "          Directories to search for the libraries can be specified in the\n",
      "          numpy/distutils/site.cfg file (section [blas]) or by setting\n",
      "          the BLAS environment variable.\n",
      "        self.calc_info()\n",
      "      blas_src_info:\n",
      "        NOT AVAILABLE\n",
      "  \n",
      "      C:\\Users\\shm22004\\AppData\\Local\\Temp\\pip-install-n2b5b22y\\numpy_a28c2754541940f18367734d5cdccbc7\\numpy\\distutils\\system_info.py:625: UserWarning:\n",
      "          Blas (http://www.netlib.org/blas/) sources not found.\n",
      "          Directories to search for the sources can be specified in the\n",
      "          numpy/distutils/site.cfg file (section [blas_src]) or by setting\n",
      "          the BLAS_SRC environment variable.\n",
      "        self.calc_info()\n",
      "        NOT AVAILABLE\n",
      "  \n",
      "      'svnversion' is not recognized as an internal or external command,\n",
      "      operable program or batch file.\n",
      "      non-existing path in 'numpy\\\\distutils': 'site.cfg'\n",
      "      lapack_opt_info:\n",
      "      lapack_mkl_info:\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries mkl_rt not found in ['C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\lib', 'C:\\\\', 'C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\libs']\n",
      "        NOT AVAILABLE\n",
      "  \n",
      "      openblas_lapack_info:\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries openblas not found in ['C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\lib', 'C:\\\\', 'C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\libs']\n",
      "        NOT AVAILABLE\n",
      "  \n",
      "      openblas_clapack_info:\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries openblas,lapack not found in ['C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\lib', 'C:\\\\', 'C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\libs']\n",
      "        NOT AVAILABLE\n",
      "  \n",
      "      atlas_3_10_threads_info:\n",
      "      Setting PTATLAS=ATLAS\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries lapack_atlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries tatlas,tatlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries lapack_atlas not found in C:\\\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries tatlas,tatlas not found in C:\\\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries lapack_atlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\libs\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries tatlas,tatlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\libs\n",
      "      <class 'numpy.distutils.system_info.atlas_3_10_threads_info'>\n",
      "        NOT AVAILABLE\n",
      "  \n",
      "      atlas_3_10_info:\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries lapack_atlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries satlas,satlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries lapack_atlas not found in C:\\\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries satlas,satlas not found in C:\\\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries lapack_atlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\libs\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries satlas,satlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\libs\n",
      "      <class 'numpy.distutils.system_info.atlas_3_10_info'>\n",
      "        NOT AVAILABLE\n",
      "  \n",
      "      atlas_threads_info:\n",
      "      Setting PTATLAS=ATLAS\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries lapack_atlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries ptf77blas,ptcblas,atlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries lapack_atlas not found in C:\\\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries ptf77blas,ptcblas,atlas not found in C:\\\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries lapack_atlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\libs\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries ptf77blas,ptcblas,atlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\libs\n",
      "      <class 'numpy.distutils.system_info.atlas_threads_info'>\n",
      "        NOT AVAILABLE\n",
      "  \n",
      "      atlas_info:\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries lapack_atlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries f77blas,cblas,atlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries lapack_atlas not found in C:\\\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries f77blas,cblas,atlas not found in C:\\\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries lapack_atlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\libs\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries f77blas,cblas,atlas not found in C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\libs\n",
      "      <class 'numpy.distutils.system_info.atlas_info'>\n",
      "        NOT AVAILABLE\n",
      "  \n",
      "      lapack_info:\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      customize MSVCCompiler\n",
      "        libraries lapack not found in ['C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\lib', 'C:\\\\', 'C:\\\\users\\\\shm22004\\\\appdata\\\\local\\\\programs\\\\python\\\\python39\\\\libs']\n",
      "        NOT AVAILABLE\n",
      "  \n",
      "      C:\\Users\\shm22004\\AppData\\Local\\Temp\\pip-install-n2b5b22y\\numpy_a28c2754541940f18367734d5cdccbc7\\numpy\\distutils\\system_info.py:625: UserWarning:\n",
      "          Lapack (http://www.netlib.org/lapack/) libraries not found.\n",
      "          Directories to search for the libraries can be specified in the\n",
      "          numpy/distutils/site.cfg file (section [lapack]) or by setting\n",
      "          the LAPACK environment variable.\n",
      "        self.calc_info()\n",
      "      lapack_src_info:\n",
      "        NOT AVAILABLE\n",
      "  \n",
      "      C:\\Users\\shm22004\\AppData\\Local\\Temp\\pip-install-n2b5b22y\\numpy_a28c2754541940f18367734d5cdccbc7\\numpy\\distutils\\system_info.py:625: UserWarning:\n",
      "          Lapack (http://www.netlib.org/lapack/) sources not found.\n",
      "          Directories to search for the sources can be specified in the\n",
      "          numpy/distutils/site.cfg file (section [lapack_src]) or by setting\n",
      "          the LAPACK_SRC environment variable.\n",
      "        self.calc_info()\n",
      "        NOT AVAILABLE\n",
      "  \n",
      "      C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\\distutils\\dist.py:274: UserWarning: Unknown distribution option: 'define_macros'\n",
      "        warnings.warn(msg)\n",
      "      running install\n",
      "      running build\n",
      "      running config_cc\n",
      "      unifing config_cc, config, build_clib, build_ext, build commands --compiler options\n",
      "      running config_fc\n",
      "      unifing config_fc, config, build_clib, build_ext, build commands --fcompiler options\n",
      "      running build_src\n",
      "      build_src\n",
      "      building py_modules sources\n",
      "      building library \"npymath\" sources\n",
      "      No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      error: Microsoft Visual C++ 14.0 or greater is required. Get it with \"Microsoft C++ Build Tools\": https://visualstudio.microsoft.com/visual-cpp-build-tools/\n",
      "      ----------------------------------------\n",
      "  ERROR: Command errored out with exit status 1: 'C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\shm22004\\\\AppData\\\\Local\\\\Temp\\\\pip-install-n2b5b22y\\\\numpy_a28c2754541940f18367734d5cdccbc7\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\shm22004\\\\AppData\\\\Local\\\\Temp\\\\pip-install-n2b5b22y\\\\numpy_a28c2754541940f18367734d5cdccbc7\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' install --record 'C:\\Users\\shm22004\\AppData\\Local\\Temp\\pip-record-ba051w6x\\install-record.txt' --single-version-externally-managed --prefix 'C:\\Users\\shm22004\\AppData\\Local\\Temp\\pip-build-env-s8hp1rbk\\overlay' --compile --install-headers 'C:\\Users\\shm22004\\AppData\\Local\\Temp\\pip-build-env-s8hp1rbk\\overlay\\Include\\numpy' Check the logs for full command output.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/92/1f/206195564030e712f85952117171df7bba7e8f945aaeb6f4cedbc731079a/pyarrow-0.17.0.tar.gz#sha256=fb1cdfda872700feee8ce8cc1f4a7e4220728d6e31eb067a0f95595f11e724f3 (from https://pypi.org/simple/pyarrow/) (requires-python:>=3.5). Command errored out with exit status 1: 'C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\python.exe' 'C:\\Users\\shm22004\\AppData\\Local\\Temp\\pip-standalone-pip-h7q1_5fr\\__env_pip__.zip\\pip' install --ignore-installed --no-user --prefix 'C:\\Users\\shm22004\\AppData\\Local\\Temp\\pip-build-env-s8hp1rbk\\overlay' --no-warn-script-location --no-binary :none: --only-binary :none: -i https://pypi.org/simple -- 'cython >= 0.29' 'numpy==1.14.5; python_version<'\"'\"'3.7'\"'\"'' 'numpy==1.16.0; python_version>='\"'\"'3.7'\"'\"'' setuptools setuptools_scm wheel Check the logs for full command output.\n",
      "ERROR: Cannot install tfx==0.13.0, tfx==0.14.0, tfx==0.15.0, tfx==0.21.0, tfx==0.21.1, tfx==0.21.2, tfx==0.21.3, tfx==0.21.4, tfx==0.21.5, tfx==0.22.0, tfx==0.22.1, tfx==0.22.2, tfx==0.23.0 and tfx==0.23.1 because these package versions have conflicting dependencies.\n",
      "ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/user_guide/#fixing-conflicting-dependencies\n"
     ]
    }
   ],
   "source": [
    "%pip install tfx"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting dart\n",
      "  Downloading dart-1.0-py2.py3-none-any.whl (3.4 kB)\n",
      "Collecting flask-sqlalchemy\n",
      "  Downloading Flask_SQLAlchemy-2.5.1-py2.py3-none-any.whl (17 kB)\n",
      "Collecting SQLAlchemy>=0.8.0\n",
      "  Downloading SQLAlchemy-1.4.26-cp39-cp39-win_amd64.whl (1.5 MB)\n",
      "Collecting Flask>=0.10\n",
      "  Downloading Flask-2.0.2-py3-none-any.whl (95 kB)\n",
      "Collecting click>=7.1.2\n",
      "  Downloading click-8.0.3-py3-none-any.whl (97 kB)\n",
      "Collecting Jinja2>=3.0\n",
      "  Downloading Jinja2-3.0.2-py3-none-any.whl (133 kB)\n",
      "Collecting itsdangerous>=2.0\n",
      "  Downloading itsdangerous-2.0.1-py3-none-any.whl (18 kB)\n",
      "Collecting Werkzeug>=2.0\n",
      "  Downloading Werkzeug-2.0.2-py3-none-any.whl (288 kB)\n",
      "Collecting greenlet!=0.4.17\n",
      "  Downloading greenlet-1.1.2-cp39-cp39-win_amd64.whl (101 kB)\n",
      "Requirement already satisfied: colorama in c:\\users\\shm22004\\appdata\\roaming\\python\\python39\\site-packages (from click>=7.1.2->Flask>=0.10->flask-sqlalchemy->dart) (0.4.4)\n",
      "Collecting MarkupSafe>=2.0\n",
      "  Downloading MarkupSafe-2.0.1-cp39-cp39-win_amd64.whl (14 kB)\n",
      "Installing collected packages: MarkupSafe, Werkzeug, Jinja2, itsdangerous, greenlet, click, SQLAlchemy, Flask, flask-sqlalchemy, dart\n",
      "  Attempting uninstall: MarkupSafe\n",
      "    Found existing installation: MarkupSafe 1.1.1\n",
      "    Uninstalling MarkupSafe-1.1.1:\n",
      "      Successfully uninstalled MarkupSafe-1.1.1\n",
      "  Attempting uninstall: Werkzeug\n",
      "    Found existing installation: Werkzeug 1.0.1\n",
      "    Uninstalling Werkzeug-1.0.1:\n",
      "      Successfully uninstalled Werkzeug-1.0.1\n",
      "  Attempting uninstall: Jinja2\n",
      "    Found existing installation: Jinja2 2.11.3\n",
      "    Uninstalling Jinja2-2.11.3:\n",
      "      Successfully uninstalled Jinja2-2.11.3\n",
      "Successfully installed Flask-2.0.2 Jinja2-3.0.2 MarkupSafe-2.0.1 SQLAlchemy-1.4.26 Werkzeug-2.0.2 click-8.0.3 dart-1.0 flask-sqlalchemy-2.5.1 greenlet-1.1.2 itsdangerous-2.0.1\n"
     ]
    }
   ],
   "source": [
    "%pip install dart"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " Volume in drive C is OS"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "File Not Found\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      " Volume Serial Number is 70C1-9391\n",
      "\n",
      " Directory of c:\\shemies\\E_Drive\\python\\jptr py\\New folder\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!dir dart"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "UsageError: Line magic function `%labextension` not found.\n"
     ]
    }
   ],
   "source": [
    "#!pip install witwidget\n",
    "#!pip install jupyter\n",
    "%labextension install wit-widget\n",
    "%labextension install @jupyter-widgets/Jupyter-manager"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: ipykernel in c:\\users\\shm22004\\appdata\\roaming\\python\\python39\\site-packages (5.5.5)\n",
      "Collecting ipykernel\n",
      "  Downloading ipykernel-6.9.1-py3-none-any.whl (128 kB)\n",
      "     ------------------------------------ 128.9/128.9 KB 631.1 kB/s eta 0:00:00\n",
      "Requirement already satisfied: nest-asyncio in c:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from ipykernel) (1.5.1)\n",
      "Collecting traitlets<6.0,>=5.1.0\n",
      "  Downloading traitlets-5.1.1-py3-none-any.whl (102 kB)\n",
      "     -------------------------------------- 102.0/102.0 KB 2.0 MB/s eta 0:00:00\n",
      "Collecting debugpy<2.0,>=1.0.0\n",
      "  Downloading debugpy-1.5.1-cp39-cp39-win_amd64.whl (4.4 MB)\n",
      "     ---------------------------------------- 4.4/4.4 MB 1.9 MB/s eta 0:00:00\n",
      "Requirement already satisfied: jupyter-client<8.0 in c:\\users\\shm22004\\appdata\\roaming\\python\\python39\\site-packages (from ipykernel) (6.1.12)\n",
      "Requirement already satisfied: ipython>=7.23.1 in c:\\users\\shm22004\\appdata\\roaming\\python\\python39\\site-packages (from ipykernel) (7.24.1)\n",
      "Requirement already satisfied: tornado<7.0,>=4.2 in c:\\users\\shm22004\\appdata\\roaming\\python\\python39\\site-packages (from ipykernel) (6.1)\n",
      "Requirement already satisfied: matplotlib-inline<0.2.0,>=0.1.0 in c:\\users\\shm22004\\appdata\\roaming\\python\\python39\\site-packages (from ipykernel) (0.1.2)\n",
      "Requirement already satisfied: pygments in c:\\users\\shm22004\\appdata\\roaming\\python\\python39\\site-packages (from ipython>=7.23.1->ipykernel) (2.9.0)\n",
      "Requirement already satisfied: prompt-toolkit!=3.0.0,!=3.0.1,<3.1.0,>=2.0.0 in c:\\users\\shm22004\\appdata\\roaming\\python\\python39\\site-packages (from ipython>=7.23.1->ipykernel) (3.0.18)\n",
      "Requirement already satisfied: colorama in c:\\users\\shm22004\\appdata\\roaming\\python\\python39\\site-packages (from ipython>=7.23.1->ipykernel) (0.4.4)\n",
      "Requirement already satisfied: pickleshare in c:\\users\\shm22004\\appdata\\roaming\\python\\python39\\site-packages (from ipython>=7.23.1->ipykernel) (0.7.5)\n",
      "Requirement already satisfied: setuptools>=18.5 in c:\\users\\shm22004\\appdata\\roaming\\python\\python39\\site-packages (from ipython>=7.23.1->ipykernel) (57.0.0)\n",
      "Requirement already satisfied: backcall in c:\\users\\shm22004\\appdata\\roaming\\python\\python39\\site-packages (from ipython>=7.23.1->ipykernel) (0.2.0)\n",
      "Requirement already satisfied: jedi>=0.16 in c:\\users\\shm22004\\appdata\\roaming\\python\\python39\\site-packages (from ipython>=7.23.1->ipykernel) (0.18.0)\n",
      "Requirement already satisfied: decorator in c:\\users\\shm22004\\appdata\\roaming\\python\\python39\\site-packages (from ipython>=7.23.1->ipykernel) (5.0.9)\n",
      "Requirement already satisfied: python-dateutil>=2.1 in c:\\users\\shm22004\\appdata\\roaming\\python\\python39\\site-packages (from jupyter-client<8.0->ipykernel) (2.8.1)\n",
      "Requirement already satisfied: jupyter-core>=4.6.0 in c:\\users\\shm22004\\appdata\\roaming\\python\\python39\\site-packages (from jupyter-client<8.0->ipykernel) (4.7.1)\n",
      "Requirement already satisfied: pyzmq>=13 in c:\\users\\shm22004\\appdata\\roaming\\python\\python39\\site-packages (from jupyter-client<8.0->ipykernel) (22.1.0)\n",
      "Requirement already satisfied: parso<0.9.0,>=0.8.0 in c:\\users\\shm22004\\appdata\\roaming\\python\\python39\\site-packages (from jedi>=0.16->ipython>=7.23.1->ipykernel) (0.8.2)\n",
      "Requirement already satisfied: pywin32>=1.0 in c:\\users\\shm22004\\appdata\\roaming\\python\\python39\\site-packages (from jupyter-core>=4.6.0->jupyter-client<8.0->ipykernel) (301)\n",
      "Requirement already satisfied: wcwidth in c:\\users\\shm22004\\appdata\\roaming\\python\\python39\\site-packages (from prompt-toolkit!=3.0.0,!=3.0.1,<3.1.0,>=2.0.0->ipython>=7.23.1->ipykernel) (0.2.5)\n",
      "Requirement already satisfied: six>=1.5 in c:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\lib\\site-packages (from python-dateutil>=2.1->jupyter-client<8.0->ipykernel) (1.15.0)\n",
      "Installing collected packages: traitlets, debugpy, ipykernel\n",
      "  Attempting uninstall: traitlets\n",
      "    Found existing installation: traitlets 5.0.5\n",
      "    Uninstalling traitlets-5.0.5:\n",
      "      Successfully uninstalled traitlets-5.0.5\n",
      "  Attempting uninstall: ipykernel\n",
      "    Found existing installation: ipykernel 5.5.5\n",
      "    Uninstalling ipykernel-5.5.5:\n",
      "      Successfully uninstalled ipykernel-5.5.5\n",
      "Successfully installed debugpy-1.5.1 ipykernel-6.9.1 traitlets-5.1.1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "WARNING: You are using pip version 22.0.3; however, version 22.0.4 is available.\n",
      "You should consider upgrading via the 'C:\\users\\shm22004\\appdata\\local\\programs\\python\\python39\\python.exe -m pip install --upgrade pip' command.\n"
     ]
    }
   ],
   "source": [
    "%pip install -U ipykernel"
   ]
  }
 ],
 "metadata": {
  "interpreter": {
   "hash": "b872ac7de5fca3a5420d01e53df10c44aa1d8c9e6194de55071935c6ef55fdcf"
  },
  "kernelspec": {
   "display_name": "Python 3.9.1 64-bit",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.1"
  },
  "orig_nbformat": 2
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
